backend/layers/thirdparty/crossref_provider.py

```diff
@@ -143,19 +146,11 @@ def fetch_metadata(self, doi: str) -> dict:
         except Exception as e:
             raise CrossrefParseException("Cannot parse metadata from Crossref") from e
 
-    def fetch_preprint_published_doi(self, doi):
-        """
-        Given a preprint DOI, returns the DOI of the published paper, if available.
-        """
-
-        res = self._fetch_crossref_payload(doi)
-        message = res.json()["message"]
-        is_preprint = message.get("subtype") == "preprint"
-
-        if is_preprint:
-            try:
-                published_doi = message["relation"]["is-preprint-of"]
-                if published_doi[0]["id-type"] == "doi":
-                    return published_doi[0]["id"]
-            except Exception:
-                pass
+    def fetch_published_metadata(self, doi_response_message: dict) -> dict:
+        try:
+            published_doi = doi_response_message["relation"]["is-preprint-of"]
+            # the new DOI to query for ...
+            if published_doi[0]["id-type"] == "doi":
+                return self.fetch_metadata(published_doi[0]["id"])
+        except Exception:  # if fetch of published doi errors out, just use preprint doi
+            pass
```

Comment:
nit: since we are expecting a return value and assigning the output elsewhere, I think its clearer to explicitly return None here instead. But that's a personal preference